### PR TITLE
Ruby on Rails capstone project Budget app

### PR DESCRIPTION
**Pull request by marilirulita**
### It allows a user to:

- Register and log in, so that the data is private to them.
- Introduce new transactions associated with a category.
- See the money spent on each category.
- It uses Postgres as a database.
- Use the device for authentication.
- The project is deployed and accessible online.
- It has unit and integration tests.

### CODE REVIEW

File: app/models/group.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Group < ApplicationRecord
+  belongs_to :user
+  has_many :group_moves, dependent: :destroy
+  has_many :moves, through: :group_moves
+
+  validates :name, presence: true
+  validates :icon, presence: true
```
Comment (smunozmo) [inline app/models/group.rb]: - [x] You've done an excellent job on this project, congrats! However, I would suggest validating the type of data for :name and :amount, for example, you could say :amount should be an integer greater than cero.

File: app/models/move.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Move < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :group_moves, dependent: :destroy
+  has_many :groups, through: :group_moves
+
+  validates :name, presence: true
+  validates :amount, presence: true
```
Comment (smunozmo) [inline app/models/move.rb]: - [x] Same here, I would recommend validating the type of data too. This avoids any error even if anyone tried to input data from the console.

File: app/views/moves/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<h1>Editing move</h1>
```
Comment (smunozmo) [inline app/views/moves/edit.html.erb:1]: - [OPTIONA] I would recommend removing unused files from the scaffold, this way your code would be cleaner and easier to maintain.

**Review by smunozmo — CHANGES_REQUESTED** [review]
Hi @marilirulita, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 👏 

- Another cool project for your portfolio, congrats on it! 🎉 
- You did an amazing job with the styling, great! 😎 
- Excellent job with the live demo and video demo, perfect! 💯 
- Tests are working properly, well done! ✅ 
- All pages are displayed correctly, nice! 👍 
- All associations are well set up, excellent! 🚀 
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! 🤓 

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

⚠️  **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/group.rb (hunk 1 above)
Comment (marilirulita) [reply app/models/group.rb]: thanks, @smunozmo I implemented it now

File: app/models/move.rb (hunk 2 above)
Comment (marilirulita) [reply app/models/move.rb]: 👍🏽

File: app/views/moves/edit.html.erb (hunk 3 above)
Comment (marilirulita) [reply app/views/moves/edit.html.erb:1]: Done! 👍🏽  thanks @smunozmo  for the review

**Review by EricMbouwe — APPROVED** [review]
Hi @marilirulita, 

In my opinion, your project is ready for presentation! ⚡🔥 There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**